state/test/test_pruning_state.py

```diff
@@ -261,6 +261,10 @@ def test_get_for_old_root(state):
     assert state.get_for_root_hash(head_hash2, b'k2') == b'v2'
     assert state.get_for_root_hash(head_hash2, b'k3') == b'v3'
 
+    assert len(state.get_all_leaves_for_root_hash(head_hash1)) == 2
```

Comment:
Please add a specific test for `get_all_leaves_for_root_hash` and check the content as well, not only length.

Reply:
ok